cmake/setup_cached_variables.cmake

```diff
@@ -129,6 +129,17 @@ IF( NOT "${CMAKE_BUILD_TYPE}" STREQUAL "Release" AND
 ENDIF()
 
 #
+# Force exactly one build type for multi-cofiguration generators.
```

Comment:
But I also don't know what this actually means. What is a multi configuration generator?